monitoring/mongodb/dashboard.py

```diff
@@ -0,0 +1,1397 @@
+        dataSource=DATASOURCE,
+        unit=unit,
+        thresholds=thresholds,
+        targets=targets,
+        **kwargs,
+    )
+
+
+def mongodb_state_timeline(
+    title, expr, description="", mappings=None, width=None, **kwargs
+):
+    """Create a MongoDB StateTimeline panel"""
+    # Set default values if not provided in kwargs
+    defaults = {
+        "fillOpacity": 100,
+        "lineWidth": 0,
+        "showValue": "auto",
+        "mergeValues": True,
+        "rowHeight": 0.9,
+        "colorMode": "palette-classic",
+    }
+
+    # Update defaults with any provided kwargs
+    for key, default_value in defaults.items():
+        if key not in kwargs:
+            kwargs[key] = default_value
+
+    panel = StateTimeline(
+        title=title,
+        description=description,
+        dataSource=DATASOURCE,
+        mappings=mappings,
+        targets=[Target(expr=expr, legendFormat="{{pod}}")],
+        **kwargs,
+    )
+    if width:
+        panel.gridPos = GridPos(h=7, w=width, x=0, y=0)
+    return panel
+
+
+# Overview Panels
+mongodb_services_state = mongodb_stat(
+    "MongoDB services state",
+    orientation="horizontal",
+    targets=[
+        Target(
+            expr=MongoDBMetrics.up(JOBS["configsvr"]),
+            legendFormat="Config Server up:",
+        ),
+        Target(
+            expr=MongoDBMetrics.up(JOBS["mongos"]),
+            legendFormat="MongoS up:",
+        ),
+        Target(
+            expr=MongoDBMetrics.up(JOBS["shards"]),
+            legendFormat="Shard Server up:",
+        ),
+    ],
+)
+
+version_panel = Stat(
+    title="Version",
+    dataSource=DATASOURCE,
+    format="string",
+    colorMode="none",
+    orientation="horizontal",
+    textMode="auto",
+    reduceOptions={"calcs": ["lastNotNull"], "fields": "/^mongodb$/", "values": False},
+    textSize=20,
+    wideLayout=True,
+    targets=[
+        Target(
+            expr=MongoDBMetrics.version_info(),
+            format="table",
+            instant=True,
+            legendFormat="{{mongodb}}",
+        )
+    ],
+)
+
+disk_usage_pie = PieChart(
+    title="Disk usage",
+    dataSource=DATASOURCE,
+    displayLabels=["value", "name", "percent"],
+    legendDisplayMode="hidden",
+    pieType="pie",
+    unit=UNITS.BYTES,
+    targets=[
+        Target(
+            expr=MongoDBMetrics.db_stats("indexSize"),
+            legendFormat="Indexes",
+        ),
+        Target(
+            expr=f'{MongoDBMetrics.db_stats("fsUsedSize")} - {MongoDBMetrics.db_stats("indexSize")}',
+            legendFormat="Documents",
+        ),
+        Target(
+            expr=f'{MongoDBMetrics.db_stats("fsTotalSize")} - {MongoDBMetrics.db_stats("fsUsedSize")}',
+            legendFormat="Free",
+        ),
+    ],
+)
+
+avg_doc_size = mongodb_stat(
+    "Average document size",
+    expr=f'avg(mongodb_dbstats_avgObjSize{{database!~"admin|config|local", rs_state="1", job="{JOBS["shards"]}"}})',
+    unit=UNITS.BYTES,
+    decimals=3,
```

Comment:
do we really need so many decimals? in the screenshot, 138.969 bytes would be better rounded simply to 140

Grafana will automatically convert to the most appropriate metric prefix (k, ...), may be enough to keep 0 or 1 decimal for the average document size?